backend/src/file/file.go

```diff
@@ -0,0 +1,159 @@
+package file
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"strings"
+
+	"github.com/GenerateNU/sac/backend/src/config"
+	"github.com/GenerateNU/sac/backend/src/errors"
+	"github.com/GenerateNU/sac/backend/src/models"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/google/uuid"
+)
+
+type FileTypes string
+
+const (
+	IMAGE       FileTypes = "image"
+	VIDEO       FileTypes = "video"
+	AUDIO       FileTypes = "audio"
+	DOCUMENT    FileTypes = "document"
+	APPLICATION FileTypes = "application"
+	TEXT        FileTypes = "text"
+	ALL         FileTypes = "all"
+)
+
+type FileClientInterface interface {
+	GetFileURL(fileURL string) *string
+	UploadFile(folder string, fileHeader *multipart.FileHeader, allowedTypes []FileTypes) (*models.FileInfo, *errors.Error)
+	DeleteFile(fileURL string) *errors.Error
+}
+
+type AWSClient struct {
+	Settings config.AWSSettings
+	session  *session.Session
+}
+
+func NewAWSProvider(settings config.AWSSettings) FileClientInterface {
+	sess, err := session.NewSession(&aws.Config{
+		Region:      aws.String(settings.REGION.Expose()),
+		Credentials: credentials.NewStaticCredentials(settings.ID.Expose(), settings.SECRET.Expose(), ""),
+	})
+	if err != nil {
+		return nil
+	}
+
+	return &AWSClient{Settings: settings, session: sess}
+}
+
+func (aw *AWSClient) GetFileURL(fileURL string) *string {
+	fileURL = fmt.Sprintf("https://%s.s3.amazonaws.com/%s", aw.Settings.BUCKET_NAME.Expose(), fileURL)
+	return &fileURL
+}
+
+func preProcessFile(fileHeader *multipart.FileHeader) (*string, []byte, *errors.Error) {
+	if fileHeader.Size > 5000000 {
```

Comment:
nit: should this configurable? does 5_000_000 make sense for our current use cases?